### PR TITLE
EsLint configuration

### PR DESCRIPTION
**Pull request by Raghav-Maheshwari**
This PR sets up EsLint and Prettier to be used across the code base and applies stylistic rules across frontend and backend files

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,18 @@
...
+    "eslint": "^6.0.0",
+    "eslint-config-airbnb": "^17.1.0",
+    "eslint-config-prettier": "^6.3.0",
+    "eslint-plugin-babel": "^5.3.0",
+    "eslint-plugin-import": "^2.17.3",
+    "eslint-plugin-jsx-a11y": "^6.2.1",
+    "eslint-plugin-prettier": "^3.1.0",
+    "eslint-plugin-react": "^7.13.0",
+    "prettier": "^1.18.2",
+    "pretty-quick": "^1.11.1"
```
Comment (lucashu1) [inline package.json:27]: Remove from root level

File: .eslintrc [hunk 2]
```diff
@@ -1,11 +1,32 @@
 {
-  "parser": "babel-eslint",
   "env": {
-    "node": true,
-    "es6": true
+    "browser": true,
```
Comment (lucashu1) [inline .eslintrc:3]: I think this should be removed from packages/backend?

File: .eslintrc [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
+    "es6": true,
+    "commonjs": true
+  },
+  "extends": ["airbnb", "prettier", "prettier/react"],
```
Comment (lucashu1) [inline .eslintrc:7]: Remove react from backend

File: .eslintrc [hunk 4]
```diff
@@ -1,11 +1,32 @@
...
+  "parser": "babel-eslint",
+  "parserOptions": {
+    "ecmaVersion": 9,
+    "ecmaFeatures": {
+      "jsx": true
+    },
+    "sourceType": "module"
   },
-  "extends": "eslint:recommended",
+  "plugins": ["react", "prettier"],
```
Comment (lucashu1) [inline .eslintrc:16]: Remove react from backend

**Comment by lucashu1** [comment]
Relevant links:
* https://github.com/microsoft/vscode-eslint/issues/696
* https://stackoverflow.com/questions/32964920/should-i-commit-the-vscode-folder-to-source-control

```
# Created by https://www.gitignore.io/api/visualstudiocode
# Edit at https://www.gitignore.io/?templates=visualstudiocode
### VisualStudioCode ###
.vscode/*
!.vscode/settings.json
!.vscode/tasks.json
!.vscode/launch.json
!.vscode/extensions.json
### VisualStudioCode Patch ###
# Ignore all local history of files
**/.history
# End of https://www.gitignore.io/api/visualstudiocode
```

**Comment by Raghav-Maheshwari** [comment]
Who knew setting up linters and formatters could be such a pain in the ass?

Anyways, I think everything should be configured and styled correctly now.
Both the frontend and backend are linted using the [airbnb style guide](https://github.com/airbnb/javascript), and formatted using Prettier.

@lucashu1 I did end up having to push a vscode settings file with the following:
```
{
  "eslint.workingDirectories": [
    "./packages/frontend",
    "./packages/backend"
  ],
}
```

And to make VS code work, I have the following user-level settings.
```
 "editor.codeActionsOnSave": {
    "source.fixAll.format": true,
    "source.fixAll.eslint": true
  },
  "editor.formatOnSave": false,
  "[javascript]": {
    "editor.formatOnSave": false
  },
  "eslint.alwaysShowStatus": true,
```

**Pull request merged**